proxy_docker/app/script/call_lightningd.sh

```diff
@@ -306,21 +324,30 @@ ln_pay() {
     local invoice_description=$(echo "${result}" | jq ".description")
     trace "[ln_pay] invoice_description=${invoice_description}"
 
-    # The amount must match
-    if [ "${expected_msatoshi}" != "${invoice_msatoshi}" ]; then
+    # The amount must match if not "any"
+    # If the amount is not in the invoice and not supplied as expected_msatoshi, then both will be null, that's ok!
+    # Same thing goes for the description.
+    if [ "${expected_msatoshi}" != "${invoice_msatoshi}" ] && [ "${invoice_msatoshi}" != "null" ]; then
+      # If invoice_msatoshi is null, that means "any" was supplied, so the amounts don't have to match!
       result="{\"result\":\"error\",\"expected_msatoshi\":${expected_msatoshi},\"invoice_msatoshi\":${invoice_msatoshi}}"
       returncode=1
-    elif [ "${expected_description}" != '""' ] && [ "${expected_description}" != "${invoice_description}" ]; then
-      # If expected description is empty, we accept any description on the invoice.  Amount is the important thing.
+    elif [ -n "${expected_description}" ] && [ "${expected_description}" != "null" ] && [ "${expected_description}" != "${invoice_description}" ]; then
+      # If expected description is empty, we accept any description on the invoice.  Amount is the most important thing.
 
       result="{\"result\":\"error\",\"expected_description\":${expected_description},\"invoice_description\":${invoice_description}}"
       returncode=1
     else
       # Amount and description is as expected, let's pay!
       trace "[ln_pay] Amount and description are as expected, let's try to pay!"
```

Comment:
trace message 
```
"[ln_pay] Amount and description are as expected, let's try to pay!"
```
does not correspond with code comment
```
# If expected description is empty, we accept any description on the invoice.  Amount is the most important thing.
```
Not sure what the callers expectation here is on invoice description matching check.

Reply:
I see what you mean :)